<?xml version="1.0"?>
<robot name="lumi" xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:include filename="$(find franka_sim)/model/robots/panda_arm.xacro"/>
    <xacro:include filename="$(find franka_sim)/model/robots/hand.xacro"/>

    <xacro:panda_arm connected_to="base_link" arm_id="lumi" xyz="-0.4 0.15 0.0"/>
    <xacro:hand ns="lumi" rpy="0 0 ${-pi/4}" connected_to="lumi_link8"/>

    <mujoco>
        <compiler meshdir="../meshes/" balanceinertia="true" discardvisual="false"/>
    </mujoco>

    <link name="base_link"/>

    <link name="door_base"/>

    <joint name="door_base_joint" type="fixed">
        <!-- <origin xyz="0.0 0.0 0.052" rpy="0 0 ${pi}"/> -->
        <origin xyz="0.4 0.25 0.052" rpy="0 0 ${-pi/2}"/>
        <parent link="base_link"/>
        <child link="door_base"/>
    </joint>

    <link name="door">
        <visual>
            <geometry>
              <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/door_v4_bin.stl"/>
            </geometry>
        </visual>
       <collision>
           <geometry>
               <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/door_v4_bin.stl"/>
           </geometry>
       </collision>
    </link>
    <joint name="door_hinge" type="revolute">
        <origin xyz="0.005 0.005 0.005" rpy="0 0 0"/>
        <limit lower="0" upper="${pi/2}" effort="10" velocity="10"/>
        <parent link="door_base"/>
        <child link="door"/>
    </joint>

    <link name="handle_core">
        <visual>
            <geometry>
              <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/handle_v4_core_bin.stl"/>
            </geometry>
        </visual>
       <collision>
           <geometry>
               <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/handle_v4_core_bin.stl"/>
           </geometry>
       </collision>
    </link>
    <joint name="handle_core" type="revolute">
        <origin xyz="0.052 -0.06 0.095" rpy="0 0 0"/>
        <limit lower="0" upper="${pi/6}" effort="10" velocity="10"/>
        <parent link="door"/>
        <child link="handle_core"/>
    </joint>
    
    <link name="handle_beam">
        <visual>
            <geometry>
              <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/handle_v4_beam_bin.stl"/>
            </geometry>
        </visual>
       <collision>
           <geometry>
               <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/handle_v4_beam_bin.stl"/>
           </geometry>
       </collision>
    </link>
    <joint name="handle_beam_joint" type="fixed">
        <origin xyz="0.015 0.0 -0.0165" rpy="0 0 0"/>
        <parent link="handle_core"/>
        <child link="handle_beam"/>
    </joint>
    <link name="handle_latch">
        <visual>
            <geometry>
              <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/handle_v4_latch_bin.stl"/>
            </geometry>
        </visual>
       <collision>
           <geometry>
               <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/handle_v4_latch_bin.stl"/>
           </geometry>
       </collision>
    </link>
    <joint name="handle_latch_joint" type="fixed">
        <origin xyz="-0.015 0.04 -0.01" rpy="0 0 0"/>
        <parent link="handle_core"/>
        <child link="handle_latch"/>
    </joint>
    <link name="frame_left">
        <visual>
            <geometry>
              <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/frame_v4_leftpillar_back_bin.stl"/>
            </geometry>
        </visual>
       <collision>
           <geometry>
               <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/frame_v4_leftpillar_back_bin.stl"/>
           </geometry>
       </collision>
    </link>
    <joint name="frame_left_joint" type="fixed">
        <origin xyz="-0.025 0.01 0.0" rpy="0 0 0"/>
        <parent link="door_base"/>
        <child link="frame_left"/>
    </joint>
    <link name="frame_left_front">
        <visual>
            <geometry>
              <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/frame_v4_leftpillar_front_bin.stl"/>
            </geometry>
        </visual>
       <collision>
           <geometry>
               <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/frame_v4_leftpillar_front_bin.stl"/>
           </geometry>
       </collision>
    </link>
    <joint name="frame_left_front_joint" type="fixed">
        <origin xyz="-0.025 0.0 0.0" rpy="0 0 0"/>
        <parent link="door_base"/>
        <child link="frame_left_front"/>
    </joint>
    <link name="frame_right">
        <visual>
            <geometry>
              <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/frame_v4_rightpillar_bin.stl"/>
            </geometry>
        </visual>
       <collision>
           <geometry>
               <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/frame_v4_rightpillar_bin.stl"/>
           </geometry>
       </collision>
    </link>
    <joint name="frame_right_joint" type="fixed">
        <origin xyz="0.267 0.0 0.0" rpy="0 0 0"/>
        <parent link="door_base"/>
        <child link="frame_right"/>
    </joint>
    <link name="frame_bottom">
        <visual>
            <geometry>
              <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/frame_v4_top_bottom_bin.stl"/>
            </geometry>
        </visual>
       <collision>
           <geometry>
               <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/frame_v4_top_bottom_bin.stl"/>
           </geometry>
       </collision>
    </link>
    <joint name="frame_bottom_joint" type="fixed">
        <origin xyz="-0.025 0.0 -0.052" rpy="0 0 0"/>
        <parent link="door_base"/>
        <child link="frame_bottom"/>
    </joint>
    <link name="frame_top">
        <visual>
            <geometry>
              <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/frame_v4_top_bottom_bin.stl"/>
            </geometry>
        </visual>
       <collision>
           <geometry>
               <mesh scale="0.001 0.001 0.001" filename="package://franka_sim/model/meshes/frame_v4_top_bottom_bin.stl"/>
           </geometry>
       </collision>
    </link>
    <joint name="frame_top_joint" type="fixed">
        <origin xyz="-0.025 0.0 0.335" rpy="0 0 0"/>
        <parent link="door_base"/>
        <child link="frame_top"/>
    </joint>

</robot>

<!-- <asset>
        <mesh file="stl/frame_v4_leftpillar_back_bin.stl" name="frame_left"/>
        <mesh file="stl/frame_v4_leftpillar_front_bin.stl" name="frame_left_front"/>
        <mesh file="stl/frame_v4_rightpillar_bin.stl" name="frame_right"/>
        <mesh file="stl/frame_v4_top_bottom_bin.stl" name="frame_top_bottom"/>
        <mesh file="stl/door_v4_bin.stl" name="door"/>
        <mesh file="stl/handle_v4_beam_bin.stl" name="handle_beam"/>
        <mesh file="stl/handle_v4_latch_bin.stl" name="handle_latch"/>
        <mesh file="stl/handle_v4_core_bin.stl" name="handle_core"/>
</asset> -->

<!-- <body pos="0 0 0">
    <body name="door" pos="5 5 5">
        <body name="handle_core" pos="52 -60 95">
            <body name="handle_beam" pos="15 0 -16.5">
            </body>
            <body name="handle_latch" pos="-15 40 -10">
            </body>
        </body>    
    </body>          
    <body name="frame_left" pos="-25 10 0">
    </body>
    <body name="frame_left_front" pos="-25 0 0">
    </body>
    <body name="frame_right" pos="267 0 0">
    </body>
    <body name="frame_bottom" pos="-25 0 -52">
    </body>
    <body name="frame_top" pos="-25 0 335">
    </body>
</body>    -->